mir_eval/display.py

```diff
@@ -0,0 +1,672 @@
+        ax = plt.gca()
+
+    if label_set is None:
+        # If we have non-empty pre-existing tick labels, use them
+        label_set = [_.get_text() for _ in ax.get_yticklabels()]
+        if not any(label_set):
+            label_set = []
+    else:
+        label_set = list(label_set)
+
+    # Put additional labels at the end, in order
+    if extend_labels:
+        ticks = label_set + sorted(set(labels) - set(label_set))
+    elif label_set:
+        ticks = label_set
+    else:
+        ticks = sorted(set(labels))
+
+    seg_def_style = dict(linewidth=1)
+
+    seg_map = dict()
+    seg_y = dict()
+
+    style = next(ax._get_patches_for_fill.prop_cycler)
+    if base is None:
+        base = np.arange(len(ticks))
+
+    if height is None:
+        height = np.ones(len(base))
+
+    for y0, yi, lab in zip(base, height, ticks):
+        seg_map[lab] = dict(label=lab)
+        seg_map[lab].update(seg_def_style)
+        seg_map[lab].update(style)
+        seg_map[lab]['facecolor'] = seg_map[lab].pop('color')
+        seg_map[lab].update(kwargs)
+
+        seg_y[lab] = (y0, yi)
+
+    seen = set()
+    for ival, lab in zip(intervals, labels):
+        if lab not in seg_map:
+            continue
+        over_lab = seg_map[lab].get('label', lab)
+
+        # If we've already seen this label, remove it
+        # This way, it only appears once in the legend
+        if over_lab in seen:
+            seg_map[lab].pop('label', None)
+
+        seen.add(over_lab)
+        ax.add_patch(Rectangle((ival[0], seg_y[lab][0]),
+                               ival[1] - ival[0],
+                               seg_y[lab][1], **seg_map[lab]))
+
+    # Draw a line separating the new labels from pre-existing labels
+    if label_set != ticks:
+        ax.axhline(len(label_set), color='k', alpha=0.5)
+
+    if tick:
+        ax.set_yticks([])
+        ax.grid('on', axis='y')
+        ax.set_ylim([0, len(ticks)])
+        ax.set_yticks(base)
+        ax.set_yticklabels(ticks, va='bottom')
+
+    __expand_limits(ax, [intervals.min(), intervals.max()], which='x')
+
+    return ax
+
+
+def hierarchy(intervals_hier, labels_hier, levels=None, **kwargs):
+    '''Plot a hierarchical segmentation
+
+    Parameters
+    ----------
+    intervals_hier : list of np.ndarray
+        A list of segmentation intervals.  Each element should be
+        an n-by-2 array of segment intervals, in the format returned by
+        :func:`mir_eval.io.load_intervals` or
+        :func:`mir_eval.io.load_labeled_intervals`.
+        Segmentations should be ordered by increasing specificity.
+
+    labels_hier : list of list-like
+        A list of segmentation labels.  Each element should
+        be a list of labels for the corresponding element in
+        `intervals_hier`.
+
+    levels : list of string
+        Each element `levels[i]` is a label for the `i`th segmentation.
+        This is typically used to denote the levels in a segment hierarchy.
+
+    kwargs:
```

Comment:
Above, you did not use a colon after kwargs in the docstrings.  Below, looks like you are.  I think Sphinx doesn't like to see a colon here, but I dunno.  Maybe just be consistent and do whichever is right.